src/lex.cpp

```diff
@@ -743,27 +776,52 @@ void lexer::parse_binary_number() {
   this->input_ = check_garbage_in_number_literal(input);
 }
 
-char8* lexer::check_garbage_in_number_literal(char8* input) {
-  char8* garbage_begin = input;
+void lexer::parse_octal_number(bool strict, bool is_0o) {
+  if (is_0o) assert(strict);
+  char8* input = this->input_;
+
+  while (this->is_octal_digit(*input)) {
+    ++input;
+  }
+
+  if (!is_0o) {
+    switch (*input) {
+    QLJS_CASE_DECIMAL_DIGIT:
+      this->input_ = input;
+      this->parse_number();
+      return;
+    }
+  }
+
+  if (is_0o) switch (*input)
+    case u8'n':
+      ++input;
+
+  const char8* garbage_begin = input;
```

Comment:
Can we reuse the check_garbage_in_number_literal function here? If not, why not?